### PR TITLE
Implement Dragon Booking Action(Reservation)

### PR DESCRIPTION
**Pull request by Dinma-Faith-zz**
**In this Milestone, I did the following:**

- When a user clicks the "Reserve dragon" button, action needs to be dispatched to update the store. You need to get the ID of 
  the reserved dragon and update the state. Remember you mustn't mutate the state. Instead, you need to return a new state 
  object with all dragons, but the selected rocket will have an extra key reserved with its value set to true. You could use a JS 
  filter() or map() to set the value of the new state - i.e.:

const newState = state.map(rocket => {
    if(dragon.id !== id) 
        return dragon;
    return { ...dragon, reserved: true };
});
Regardless of which method you choose, make sure you place all your logic in the reducer. In the React view file, you should only dispatch the action with the correct dragon ID as an argument.

### CODE REVIEW

**Review by ichala — APPROVED** [review]
### Hi @Dinma-Faith ,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

cr-approve


## Highlights

✔️  No linter errors
✔️ Used Git Flow
✔️ State change perfectly when click on reserve

Cheers and Happy coding!👏👏👏

File: src/Components/Dragon/Dragon.js [hunk 1]
```diff
@@ -1,20 +1,26 @@
...
+      {dragonsList.map((dragonDetail) => (
         <div key={dragonDetail.id} className="dragonPage">
-          <h3>{dragonDetail.name}</h3>
-          <p>{dragonDetail.type}</p>
-          <div className="dragon">
-            {dragonDetail.flickr_images.map((images) => (<img alt="dragonImg" key={generate()} src={images} />))}
-            <button type="button">reserve dragon</button>
-            <button type="button">cancel reservation </button>
+          <div className="dragon"><img src={dragonDetail.flickr_images} alt="dragon" /></div>
+          <div cnlassName="dragonContent">
```
Comment (jmoseka) [inline src/Components/Dragon/Dragon.js:19]: Kindly change to the correct format

**Review by jmoseka — COMMENTED** [review]
The project is ready for merge.

Optional: Kindly make changes to the following line 19 to the correct format "className"

File: src/Components/Dragon/Dragon.js (hunk 1 above)
Comment (Dinma-Faith-zz) [reply src/Components/Dragon/Dragon.js:19]: Okay @jmoseka , sure I will.
Thanks for the comment

**Pull request merged**